### PR TITLE
[OMP] Upgrade process  fails

### PR DESCRIPTION
**Pull request by withanage**
Hi @asmecher ,

Upgrading a OMP 3.1.2 database to 3.2.1 produced following errors.

- If  context is not set, the [getSupportedFormLocales()](https://github.com/pkp/omp/blob/master/classes/install/Upgrade.inc.php#L799) fails
- Depreciated [setting_type](https://github.com/pkp/omp/blob/master/classes/install/Upgrade.inc.php#L809)   in submission_settings threw errors and the upgrade script failed.

This fix should resolve those issues.

### CODE REVIEW

File: classes/install/Upgrade.inc.php [hunk 1]
```diff
@@ -796,29 +797,30 @@ function migrateSubmissionCoverImages() {
 			);
 
 			// Create a submission_settings entry for each locale
-			foreach ($contexts[$row['context_id']]->getSupportedFormLocales() as $localeKey) {
-				$newCoverPathInfo = pathinfo($newCoverPath);
-				$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /* @var $submissionDao SubmissionDAO */
-				$submissionDao->update(
-					'INSERT INTO submission_settings
+			if(isset($context)) {
```
Comment (asmecher) [inline classes/install/Upgrade.inc.php:800]: Was this possibly a left-over database entry for a cover image whose context (press) had been deleted? I don't think there's a legitimate case where the context shouldn't be available.

File: classes/install/Upgrade.inc.php [hunk 2]
```diff
@@ -796,29 +797,30 @@ function migrateSubmissionCoverImages() {
...
+					$submissionDao = DAORegistry::getDAO('SubmissionDAO');
+					/* @var $submissionDao SubmissionDAO */
+					$submissionDao->update(
+						'INSERT INTO submission_settings
 						SET
 							submission_id = ?,
 							setting_name = ?,
 							setting_value = ?,
-							locale = ?,
-							setting_type= ?',
```
Comment (asmecher) [inline classes/install/Upgrade.inc.php:809]: Good catch, thanks!

File: classes/install/Upgrade.inc.php (hunk 1 above)
Comment (withanage) [reply classes/install/Upgrade.inc.php:800]: @asmecher 
yes, this was from an old  database, where it was left-over.
I don't think, it also occurs in a real use case. Should I revert it ?

Comment (asmecher) [reply classes/install/Upgrade.inc.php:800]: It doesn't do any harm; I'm OK to leave it, just wondering where it was from.

Comment (withanage) [reply classes/install/Upgrade.inc.php:800]: sure, then let us leave it.

**Comment by asmecher** [comment]
Merged and cherry-picked to `master`. Thanks, @withanage!

**Pull request merged**